data/provider/src/main/java/tech/pegasys/teku/api/ConfigProvider.java

```diff
@@ -92,7 +99,82 @@ public GetSpecResponse getConfig() {
+        SerializableTypeDefinition.object(ConfigProvider.class);
+
+    final SpecConfig config =
+        spec
+            // Display genesis spec, for now
+            .atEpoch(UInt64.ZERO)
+            .getConfig();
+
+    config
+        .getRawConfig()
+        .forEach((k, v) -> builder.withField(k, STRING_TYPE, specConfig -> formatValue(v)));
+
+    // For the time being, manually add legacy constants for compatibility reasons
+    // These constants are no longer defined in newer config files, but may be required by consumers
+    builder
+        .withField("BLS_WITHDRAWAL_PREFIX", BYTES_TYPE, provider -> config.getBlsWithdrawalPrefix())
+        .withField(
+            "TARGET_AGGREGATORS_PER_COMMITTEE",
+            STRING_TYPE,
+            provider -> Integer.toString(ValidatorConstants.TARGET_AGGREGATORS_PER_COMMITTEE, 10))
+        .withField(
+            "RANDOM_SUBNETS_PER_VALIDATOR",
+            STRING_TYPE,
+            provider -> Integer.toString(ValidatorConstants.RANDOM_SUBNETS_PER_VALIDATOR, 10))
+        .withField(
+            "EPOCHS_PER_RANDOM_SUBNET_SUBSCRIPTION",
+            STRING_TYPE,
+            provider ->
+                Integer.toString(ValidatorConstants.EPOCHS_PER_RANDOM_SUBNET_SUBSCRIPTION, 10))
+        .withField("DOMAIN_BEACON_PROPOSER", BYTES4_TYPE, specConfig -> Domain.BEACON_PROPOSER)
+        .withField("DOMAIN_BEACON_ATTESTER", BYTES4_TYPE, specConfig -> Domain.BEACON_ATTESTER)
+        .withField("DOMAIN_RANDAO", BYTES4_TYPE, specConfig -> Domain.RANDAO)
+        .withField("DOMAIN_DEPOSIT", BYTES4_TYPE, specConfig -> Domain.DEPOSIT)
+        .withField("DOMAIN_VOLUNTARY_EXIT", BYTES4_TYPE, specConfig -> Domain.VOLUNTARY_EXIT)
+        .withField("DOMAIN_SELECTION_PROOF", BYTES4_TYPE, specConfig -> Domain.SELECTION_PROOF)
+        .withField(
+            "DOMAIN_AGGREGATE_AND_PROOF", BYTES4_TYPE, specConfig -> Domain.AGGREGATE_AND_PROOF);
+
+    // Manually add legacy altair constants
+    config
+        .toVersionAltair()
+        .ifPresent(
+            altairConfig -> {
+              builder
+                  .withField(
+                      "DOMAIN_SYNC_COMMITTEE", BYTES4_TYPE, specConfig -> Domain.SYNC_COMMITTEE)
+                  .withField(
+                      "DOMAIN_SYNC_COMMITTEE_SELECTION_PROOF",
+                      BYTES4_TYPE,
+                      specConfig -> Domain.SYNC_COMMITTEE_SELECTION_PROOF)
+                  .withField(
+                      "DOMAIN_CONTRIBUTION_AND_PROOF",
+                      BYTES4_TYPE,
+                      specConfig -> Domain.CONTRIBUTION_AND_PROOF)
+                  .withField(
+                      "TARGET_AGGREGATORS_PER_SYNC_SUBCOMMITTEE",
+                      STRING_TYPE,
+                      specConfig ->
+                          Integer.toString(
+                              ValidatorConstants.TARGET_AGGREGATORS_PER_SYNC_SUBCOMMITTEE, 10))
```

Comment:
i wonder if this would be neater if we had something like
```
private String getAggregatorsPerSyncCommittee() {
    return Integer.toString(ValidatorConstants.TARGET_AGGREGATORS_PER_SYNC_SUBCOMMITTEE, 10);
}
```
and then could just use `(__) -> getAggregatorsPerSyncCommittee()`
or make it take whatever input and just use a fn ref... maybe have a play?

Reply:
Updated to use `Integer.toString(int i)`, since radix is default 10. But moving this to another method does not help.

Reply:
i'd do a specific getter for each field if anything like above...